exporters/tests/integration/test_committime.py

```diff
@@ -0,0 +1,100 @@
+from typing import Optional
+
+import attr
+import pytest
+from committime import CommitMetric
+from committime.collector_github import GitHubCommitCollector
+from kubernetes.client import ApiClient, Configuration
+from openshift.dynamic import DynamicClient
+
+# export GIT_USER=gituser
+# export GIT_TOKEN=gittoken
+# export GIT_API=localhost:3000
+# export LOG_LEVEL=DEBUG
+# export NAMESPACES=basic-nginx-build,basic-nginx-dev,basic-nginx-stage,basic-nginx-prod
+# export TLS_VERIFY=False
+
+
+def setup_collector():
+    kube_config = Configuration()
+    kube_config.host = "https://localhost:3000"
+    kube_config.verify_ssl = False
+    dyn_client = DynamicClient(ApiClient(configuration=kube_config))
+
+    username = "gituser"
+    token = "gittoken"
+    git_api = "localhost:3000"
+    namespaces = (
+        "basic-nginx-build,basic-nginx-dev,basic-nginx-stage,basic-nginx-prod".split(
+            ","
+        )
+    )
+    apps = None
+    tls_verify = False
+    return GitHubCommitCollector(
+        dyn_client, username, token, namespaces, apps, git_api, tls_verify
+    )
+
+
+def expected_commits() -> list[CommitMetricEssentials]:
+    metrics = [
+        CommitMetricEssentials(
+            commit_timestamp=1619381788.0,
+            namespace="basic-nginx-build",
+            name="basic-nginx",
+            commit_hash="15dedb60b6208aafdfb2328a93543e3d94500978",
+            image_hash="sha256:c1282f65b5c327db4dcc6cdfb27e91338bd625d119d9ae769318f089d82e35e2",
+        ),
+        CommitMetricEssentials(
+            commit_timestamp=1619381788.0,
+            namespace="basic-nginx-build",
+            name="basic-nginx",
+            commit_hash="15dedb60b6208aafdfb2328a93543e3d94500978",
+            image_hash="sha256:4a20c8cfa48af3a938462e9cd7bfa0b16abfbc6ba16f0999f3931c79b1130e4b",
+        ),
+        CommitMetricEssentials(
+            commit_timestamp=1620401174.0,
+            namespace="basic-nginx-build",
+            name="basic-nginx",
+            commit_hash="620ce8b570c644338ba34224fc09b2d8a30bca02",
+            image_hash="sha256:71309995e6da43b76079a649b00e0aa8378443e72f1fccc76af0d73d67a7f644",
+        ),
+    ]
+    metrics.sort(key=lambda commit: commit.commit_timestamp)
+    return metrics
+
+
+@attr.define
+class CommitMetricEssentials:
+    name: str = attr.field()
+    namespace: Optional[str] = attr.field(default=None, kw_only=True)
+
+    commit_hash: Optional[str] = attr.field(default=None, kw_only=True)
+    commit_timestamp: Optional[float] = attr.field(default=None, kw_only=True)
+
+    image_hash: Optional[str] = attr.field(default=None, kw_only=True)
+
+    @staticmethod
+    def from_commit_metric(cm: CommitMetric) -> CommitMetricEssentials:
+        args = {
+            key: getattr(cm, key)
+            for key in "name namespace commit_hash commit_timestamp image_hash".split()
+        }
+
+        return CommitMetricEssentials(**args)
+
+
+@pytest.mark.integration
+def test_github_provider():
```

Comment:
this will result in a single "test" per provider. How will be grow this to account for various scenarios such as:

* Commit data from a buildConfig
* Commit data from a Tekton Pipeline
* Handling of commits that might be missing data

etc. It looks like this structure only allows for us to add more data to `expected_commits()`, but in the test results we end up missing context, as they are all lumped into one single test that passes or fails, and we won't know why. Is now the right time to expand this framework, or is that a followup?

Reply:
Since we only have mock data for the GitHub provider, this is all we have to work off of. Generalizing this _should_ happen, but later-- premature generalization is the root of all evil, or something like that.

That's part of the work I'm doing right now, actually-- as part of making a GitLab integration test next, I wanted to see if we could standardize on a set of data / pipeline actions / etc to make developing these tests easier.

Reply:
ok, I get it now. I was thrown by the name of this PR. this isn't integration tests for the entire committime exporter, just for the github provider. With that being the scope of this, then I think i'm fine as-is.